### PR TITLE
[ingress-nginx] preventing istio-proxy from terminating before the ingress-controller is stopped  

### PR DESCRIPTION
**Pull request by alexey-gavrilov-flant**
## Description
Preventing istio-proxy from terminating before the ingress-controller is stopped during grace period.
The [d8-hold-istio-proxy-termination-until-application-stops](https://deckhouse.io/documentation/v1/modules/110-istio/examples.html#experimental-feature-prevent-istio-proxy-from-terminating-before-the-main-applications-connections-are-closed) is not experimental anymore. 

## Why do we need it, and what problem does it solve?
Close https://github.com/deckhouse/deckhouse/issues/8035
Also, it is important to have working istio-proxy while controller handles connections during grace period.

## Why do we need it in the patch release (if we do)?
causes short-term inaccessibility when rolling ingress.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: fix 
summary: Preventing istio-proxy from terminating before the ingress-controller is stopped during grace period.
impact: Ingress pods will restart if `enableIstioSidecar` parameter is enabled.
impact_level: high 
```

### CODE REVIEW

File: modules/402-ingress-nginx/templates/controller/controller.yaml [hunk 1]
```diff
@@ -127,6 +127,9 @@ spec:
       annotations:
         traffic.sidecar.istio.io/includeInboundPorts: ""
         traffic.sidecar.istio.io/includeOutboundIPRanges: {{ $context.Values.global.discovery.serviceSubnet | quote }}
+      {{- if $crd.spec.enableIstioSidecar }}
+        inject.istio.io/templates: sidecar,d8-hold-istio-proxy-termination-until-application-stops
+      {{- end }}
```
Comment (miklezzzz) [inline modules/402-ingress-nginx/templates/controller/controller.yaml:132]:
```suggestion
        inject.istio.io/templates: sidecar,d8-hold-istio-proxy-termination-until-application-stops
```

Comment (miklezzzz) [reply modules/402-ingress-nginx/templates/controller/controller.yaml:132]: There is already       {{- if $crd.spec.enableIstioSidecar }} several lines above.

**Review by miklezzzz — APPROVED** [review]

File: modules/402-ingress-nginx/templates/controller/controller.yaml (hunk 1 above)
Comment (miklezzzz) [reply modules/402-ingress-nginx/templates/controller/controller.yaml:132]: image
doesn't look like Label section at all

Comment (alexey-gavrilov-flant) [reply modules/402-ingress-nginx/templates/controller/controller.yaml:132]: Yeah. It's an annotation. It's supposed to be.

Comment (miklezzzz) [reply modules/402-ingress-nginx/templates/controller/controller.yaml:132]: and why would we need having `$crd.spec.enableIstioSidecar` checked twice, before `annotations:` section and inside it?

File: modules/402-ingress-nginx/templates/controller/controller.yaml [hunk 2]
```diff
@@ -127,6 +127,7 @@ spec:
       annotations:
         traffic.sidecar.istio.io/includeInboundPorts: ""
         traffic.sidecar.istio.io/includeOutboundIPRanges: {{ $context.Values.global.discovery.serviceSubnet | quote }}
+        inject.istio.io/templates: sidecar,d8-hold-istio-proxy-termination-until-application-stops
```
Comment (apolovov) [inline modules/402-ingress-nginx/templates/controller/controller.yaml:130]:
```suggestion
        inject.istio.io/templates: "sidecar,d8-hold-istio-proxy-termination-until-application-stops"
```

**Review by apolovov — APPROVED** [review]

File: modules/110-istio/docs/EXAMPLES_RU.md [hunk 3]
```diff
@@ -620,9 +620,15 @@ spec:
 ### [экспериментальная функция] Предотвратить завершение работы istio-proxy до завершения соединений основного приложения
```
Comment (apolovov) [inline modules/110-istio/docs/EXAMPLES_RU.md:620]:
```suggestion
### Предотвратить завершение работы istio-proxy до завершения соединений основного приложения
```

**Review by yalosev — APPROVED** [review]

**Comment by yalosev** [comment]
Shouldn't we backport this?

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8109](https://github.com/deckhouse/deckhouse/pull/8109) to the branch [release-1.59](https://github.com/deckhouse/deckhouse/tree/release-1.59) successful!

**Pull request merged**